internal/client/api_version.go

```diff
@@ -160,6 +160,16 @@ func EncodeRequest(req *tikvrpc.Request) (*tikvrpc.Request, error) {
 		r := *req.RawCompareAndSwap()
 		r.Key = EncodeV2Key(ModeRaw, r.Key)
 		newReq.Req = &r
+	case tikvrpc.CmdRawChecksum:
+		r := *req.RawChecksum()
+		oriRanges := r.Ranges
+		r.Ranges = make([]*kvrpcpb.KeyRange, 0, len(oriRanges))
+		for i := 0; i < len(oriRanges); i++ {
+			keyRange := kvrpcpb.KeyRange{}
+			keyRange.StartKey, keyRange.EndKey = EncodeV2Range(ModeRaw, oriRanges[i].StartKey, oriRanges[i].EndKey)
```

Comment:
It seems that we can overwrite the `r.Ranges` here, just like how `case tikvrpc.CmdRawScan` do, to avoid allocating another slice.

Reply:
This `req` could be reencoded if we use the slice in place since the upper code might do a retry.

Reply:
OK... Then I think it worth to add a method `EncodeV2Ranges`, and add comments to warn that slice in request should not be reused.

Reply:
Make sense.

Reply:
`EncodeV2KeyRange` is added to encode the `KeyRange`